### PR TITLE
Add Wiki Gold Corpus

### PR DESCRIPTION
**Pull request by Ayushk4**
- [X] WikiGold
- [X] Docs
- [X] Tests

### CODE REVIEW

File: src/types.jl [hunk 1]
```diff
@@ -89,12 +89,21 @@ struct NERTaggedWord <: TaggedWord
     end
 end
 
+struct WikiGoldWord <: TaggedWord
```
Comment (oxinabox) [inline src/types.jl]: Maybe `NerOnlyTaggedWord` ?
We need to rethink these types at some point

Comment (Ayushk4) [reply src/types.jl]: That's better. For the time being, let's change it to `NerOnlyTaggedWord`.

File: src/WikiGold.jl [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    get_tagged(line) = push!(sent, parse_WikiGold_tagged_word(line))
+
+    new_document()
+    new_sentence()
+    # parse
+    for line in eachline(filename)
+        if length(line) == 0
+            new_sentence()
+        elseif startswith(strip(line), "-DOCSTART-")
+            length(docs) > 0 && isempty(doc[end]) && deleteat!(doc, lastindex(doc))
```
Comment (oxinabox) [inline src/WikiGold.jl]: Add comment explaining why we need to delete this?

File: src/WikiGold.jl [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            new_document()
+        else
+            get_tagged(line)
+        end
+    end
+    isempty(vcat(docs[end]...)) && deleteat!(docs, lastindex(docs))
```
Comment (oxinabox) [inline src/WikiGold.jl]: `all(isempty, docs [end])` rather than `vcat`ing them.

Also add comment about why we need this deleteat

**Pull request merged**